src/vm/ngenhash.inl

```diff
@@ -1263,8 +1263,11 @@ DPTR(VALUE) NgenHashTable<NGEN_HASH_ARGS>::FindVolatileEntryByHash(NgenHashValue
     // Since there is at least one entry there must be at least one bucket.
     _ASSERTE(m_cWarmBuckets > 0);
 
+    // Compute which bucket the entry belongs in based on the hash.
+    DWORD dwBucket = iHash % m_cWarmBuckets;
```

Comment:
This fix is very fragile. The C/C++ compilers are free to reorder the reads back to what they were before the fix. It is allowed by the C/C++ spec and we have seen the compilers to do that in number of cases.

The fix needs to use VolatileLoad or something similar.

Reply:
Ok. Using VolatileLoad is a good idea then. I was under the impression it would be an overkill